library/certificate_request.py

```diff
@@ -100,6 +112,7 @@ def _get_argument_spec():
             ca=dict(type="str", required=True),
             directory=dict(type="str", default="/etc/pki/tls"),
             provider=dict(type="str", default="certmonger"),
+            principal=dict(type="list"),
```

Comment:
I probably should have asked this earlier, but when you declare a module parameter as type `list`, then pass a string to ti as `principal: HTTP/www.example.com@EXAMPLE.com` - what happens?  Is it used in the module as a list with 1 string element?

Reply:
Exactly. The module takes care of changing it to a list with a single string on it.